### ISSUE TITLE
How to specify custom image by ID when creating machine

### ISSUE THREAD

**Issue by nilo19** [kubernetes-sigs/cluster-api-provider-azure#330]
The comment in code says `If specifying an image by ID, only the ID field needs to be set.`

But when I set image.ID only in yaml it complains that

```
spec.image.offer in body is required
spec.image.publisher in body is required
spec.image.sku in body is required
spec.image.version in body is required
```

**Comment by nilo19** [comment]
@justaugustus  do you know what's the config should be if I want to use the custom image by ID?

**Comment by devigned** [comment]
@nilo19 what version of the provider components are you using? For example:
- us.gcr.io/k8s-artifacts-prod/capi-kubeadm/cluster-api-kubeadm-controller:v0.1.5
- us.gcr.io/k8s-artifacts-prod/cluster-api/cluster-api-controller:v0.2.7
- gcr.io/k8s-staging-cluster-api-azure/cluster-api-azure-controller:latest

With the versions above, I was able to create a machine deployment with an AzureMachineTemplate like the one below:
``` yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: AzureMachineTemplate
metadata:
  name: capi-quickstart-node
spec:
  template:
    spec:
      location: southcentralus
      vmSize: Standard_B2ms
      image:
        id: "https://imagereference.microsoft.com/"
      osDisk:
        osType: "Linux"
        diskSizeGB: 30
        managedDisk:
          storageAccountType: "Premium_LRS"
      sshPublicKey: ...
```

Via the capz controller, obviously Azure API wasn't happy about the image.id reference to a fake image id as evidenced by via below, but it does show the image id was accepted as part of the spec and then relayed to the Azure API.

```
I1121 15:03:15.810673       1 azuremachine_reconciler.go:133] controllers/AzureMachine "msg"="VM does not have an id" "azureCluster"="capz-example-5f60aca8" "azureMachine"="capi-quickstart-node-6ns4x" "cluster"="capz-example-5f60aca8" "machine"="capi-quickstart-deploy-7869cd646f-v2wcb" "namespace"="default"
E1121 15:03:17.598305       1 controller.go:218] controller-runtime/controller "msg"="Reconciler error" "error"="failed to create AzureMachine VM: failed to create vm capi-quickstart-node-6ns4x : failed to create or get machine: cannot create vm: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code=\"LinkedInvalidPropertyId\" Message=\"Property id 'https://imagereference.microsoft.com/' at path 'properties.storageProfile.imageReference.id' is invalid. Expect fully qualified resource Id that start with '/subscriptions/{subscriptionId}' or '/providers/{resourceProviderNamespace}/'.\""  "controller"="azuremachine" "request"={"Namespace":"default","Name":"capi-quickstart-node-6ns4x"}
```

**Comment by nilo19** [comment]
kubeadm: v0.1.5
capz-controller: v0.2.7
capz-controller: (master) make release-manifests && kubectl apply -f infrastructure-components.yaml

**Comment by nilo19** [comment]
```yaml
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: AzureMachine
metadata:
  name: capi-test-controlplane-0
spec:
  image:
    id: ${a-valid-azure-image-id}
  location: eastus
  osDisk:
    diskSizeGB: 30
    managedDisk:
      storageAccountType: Premium_LRS
    osType: Linux
  sshPublicKey: ${SSH_PUBLIC_KEY}
  vmSize: Standard_B2ms
```

**Comment by nilo19** [comment]
I follow the step in https://cluster-api.sigs.k8s.io/user/quick-start.html and generate the `infrastructure-components.yaml ` by `make release-manifests` in master. Tell me if something goes wrong.

**Comment by CecileRobertMichon** [comment]
/close 
This was fixed by the 0.3.0 release so the quickstart should now be accurate.
